### PR TITLE
[SEMI MODULAR] Makes traitors start with 10 TC, submit opfor for the full 20.

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

This PR makes traitors start with 10 telecrystals, and requires them to submit an opposing force application and have it approved for the full 20 tc. Currently, untested but will be shortly. 

## How This Contributes To The Skyrat Roleplay Experience

Less murder/stealth, more RP.

**This is a host request. So, let's TM this when it's ready and approved.**

## Changelog

:cl:
balance: Traitors now need to submit an opposing force application for the full 20 TC. /:cl:

### CODE REVIEW

**Comment by azrael-x86** [comment]
>full 20

guh?

**Comment by hugmeorelse** [comment]
I have requested this. Please prioritize this. **Do not close this PR.**

**Comment by Arturlang** [comment]
Code changelog tag is wrong, balance is more suitable for this, given how big a change it is, there should be some text in the antag UI description and a proper changelog entry instead of the one used for refactors

File: modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm [hunk 1]
```diff
@@ -436,10 +436,19 @@
 	if(ishuman(target))
 		var/mob/living/carbon/human/human = target
 		human.put_in_hands(spawned_box)
+	if(target.mind.has_antag_datum(/datum/antagonist/traitor))
+		var/datum/component/uplink/uplink = target.mind.find_syndicate_uplink(TRUE)
```
Comment (jjpark-kb) [inline modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm:450]: have a check, what if they don't have an uplink?

File: modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm [hunk 2]
```diff
@@ -436,10 +436,19 @@
...
+		var/obj/item/stack/telecrystals = create_tc()
+		uplink.load_tc(user, telecrystals)
 
 	add_log(user.ckey, "Issued gear")
 	send_system_message("[user ? get_admin_ckey(user) : "The OPFOR subsystem"] has issued all approved equipment")
-	equipment_issued = TRUE
+
+/datum/opposing_force/proc/create_tc()
```
Comment (jjpark-kb) [inline modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm]: needs documentation

File: modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm (hunk 1 above)
Comment (magatsuchi) [reply modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm:440]: > have a check, what if they don't have an uplink?

gib()

File: modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm (hunk 2 above)
Comment (nevimer) [reply modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm]: it's a magic number right now but ima attach it to something

File: modular_skyrat/modules/opposing_force/code/items.dm [hunk 3]
```diff
@@ -4,7 +4,7 @@
 	name = "old radio"
 	desc = "A dusty and old looking radio."
 
-/obj/item/uplink/opfor/Initialize(mapload, owner, tc_amount = 0)
+/obj/item/uplink/opfor/Initialize(mapload, owner, tc_amount = 10)
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/opposing_force/code/items.dm]: why

**Comment by desertmote** [comment]
i've done the TC number changes in the PR listed below this comment. I'd like to ask you to help me add the OPFOR changes to it, or remove the non-modular TC number changes here and continue adding exclusively the OPFOR change.

**Comment by Gandalf2k15** [comment]
This is being done modularly https://github.com/Skyrat-SS13/Skyrat-tg/pull/11646

**Comment by nevimer** [comment]
https://github.com/Skyrat-SS13/Skyrat-tg/pull/11646 Is this and more, efforts focused on that PR now.

**Comment by nevimer** [comment]
okay 11646 got abandoned, this was requested to be re-opened.

**Comment by hugmeorelse** [comment]
Let's get this in the game already. It's been a week, and I've asked for this to be prioritized. Let's go already.

File: modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm [hunk 4]
```diff
@@ -445,11 +446,38 @@
 	if(ishuman(target))
 		var/mob/living/carbon/human/human = target
 		human.put_in_hands(spawned_box)
+	if(target.mind.has_antag_datum(/datum/antagonist/traitor))
+		var/datum/component/uplink/uplink = target.mind.find_syndicate_uplink(TRUE)
+		var/obj/item/stack/telecrystals =  new /obj/item/stack/telecrystal(src)
+		telecrystals.amount += TELECRYSTALS_OPFOR_BONUS
+		uplink.load_tc(user, telecrystals)
```
Comment (Zonespace27) [inline modular_skyrat/modules/opposing_force/code/opposing_force_datum.dm]: will runtime if no uplink

File: code/modules/uplink/uplink_devices.dm [hunk 5]
```diff
@@ -66,19 +65,18 @@
 	name = "dusty radio"
 	desc = "A dusty looking radio."
 
-/obj/item/uplink/old/Initialize(mapload, owner, tc_amount = 10)
+/obj/item/uplink/old/Initialize(mapload, owner, tc_amount = 20)
```
Comment (GoldenAlpharex) [inline code/modules/uplink/uplink_devices.dm:68]: Was that changed by upstream? Or was that modified non-modularly?

File: modular_skyrat/modules/skyrat-uplinks/code/uplink.dm [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	minutes_per_telecrystal = 30 // Every 30 minutes the reward raises by 1 TC
+
+	. = ..()
+
+	if(istype(target_item, /datum/objective_item/steal/hoslaser))
+		telecrystal_reward = 3
+	else if(istype(target_item, /datum/objective_item/steal/caplaser))
+		telecrystal_reward = 2
+
+	if(!telecrystal_reward)	// No reward is unfun
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/skyrat-uplinks/code/uplink.dm:50]: Why do we make it so *every* objective gives TC? Wasn't that the point, that some didn't give any, but gave more rep?

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Comment by Gandalf2k15** [comment]
No longer required, alternative methods implemented

**Pull request closed without merging**